### PR TITLE
Refactor the Sampler/Composite docs

### PR DESCRIPTION
**Pull request by arcondello**
Refactor the online documentation for samplers and composites.

### CODE REVIEW

File: dimod/reference/samplers/random_sampler.py [hunk 1]
```diff
@@ -25,22 +25,14 @@
...
-        ...
-        >>> response = dimod.RandomSampler().sample_qubo({(0, 0): -1, (1, 1): -1, (0, 1): 2}, num_reads=5)
-        >>> len(response)
-        5
-        >>> print(next(response.data()))      # doctest: +SKIP
-        Sample(sample={0: 1, 1: 0}, energy=-1.0)
+    parameters = None
+    """dict: Keyword arguments accepted by the sampling methods.
 
+    Contents are exactly `{'num_reads': []}`
```
Comment (JoelPasvolsky) [inline dimod/reference/samplers/random_sampler.py]: ?

File: dimod/reference/samplers/simulated_annealing.py [hunk 2]
```diff
@@ -38,16 +39,18 @@ class SimulatedAnnealingSampler(Sampler):
...
+        >>> J = {('a', 'b'): -1.5}
+        >>> sampleset = dimod.SimulatedAnnealingSampler().sample_ising(h, J)
 
     """
     properties = None
+
     parameters = None
+    """dict: Keyword arguments accepted by the sampling methods.
+
+    Contents are exactly `{'beta_range': [], num_reads': [], 'num_sweeps': []}`
```
Comment (JoelPasvolsky) [inline dimod/reference/samplers/simulated_annealing.py]: Will it be clear to our users that [] might not be a list?

Comment (arcondello) [reply dimod/reference/samplers/simulated_annealing.py]: In this case it is always an empty list, that's what 'Contents are exactly' means

File: dimod/reference/samplers/random_sampler.py (hunk 1 above)
Comment (arcondello) [reply dimod/reference/samplers/random_sampler.py]: See below

File: dimod/core/structured.py [hunk 3]
```diff
@@ -86,68 +86,21 @@ class Structured:
...
-
-                class TwoVariablesSampler(dimod.Sampler, dimod.Structured):
-                    @property
-                    def edgelist(self):
-                        return [(0, 1)]
-
-                    # Remaining properties and code for the sampler
-
+        """list: Edges/interactions allowed by the sampler in the form
+        `[(u, v), ...]`
```
[the remark this answers is not in the document]
Comment (arcondello) [reply dimod/core/structured.py]: '`'s make it prettier!

File: dimod/core/polysampler.py [hunk 4]
```diff
@@ -60,40 +60,55 @@ def sample_hising(self, h, J, **kwargs):
...
 
         Convert the given higher-order Ising model to a :obj:`.BinaryPolynomial`
-        and invoke `sample_poly`.
+        and then calls :meth:`.sample_poly`.
 
         Args:
             h (dict):
-                The variable biases of the Ising problem.
+                The variable biases of the Ising problem. Should be a dict of
+                the form `{v: bias, ...}` where `v` is a variable in the
```
[the remark this answers is not in the document]
Comment (arcondello) [reply dimod/core/polysampler.py]: The variable must be in {-1, 1}

File: dimod/core/structured.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoelPasvolsky) [reply dimod/core/structured.py]: They are there but were used by Github -- my point was the point or rather the period at the end that was missing

**Pull request merged**